h2o-docs/src/product/api-changes.rst

```diff
@@ -6,7 +6,38 @@ H2O-3 does its best to keep backwards compatibility between major versions, but
 From 3.32.0.1
 ~~~~~~~~~~~~~
 
-The deprecated``h2o-scala`` module has been removed.
+Modules
+'''''''
+
+The deprecated ``h2o-scala`` module has been removed.
+
+Target Encoding
+'''''''''''''''
+
+The Target Encoder API has been clarified and its consistency across clients has been improved:
+
+The following parameters are now deprecated in all clients and officially replaced by their new alternative:
+- ``k`` -> ``inflection_point``
+- ``f`` -> ``smoothing``
+- ``noise_level`` -> ``noise``
+- ``use_blending`` (R only) -> ``blending``
+
+Legacy client code using the deprecated parameters should expect a deprecation warning when using them, they are strongly encouraged to update their code to use the new naming.
+
+In an objective of performance optimization on the backend, and of simplification of the API, the ``transform`` method used to apply target encoding was modified as follow.
+- The R ``h2o.transform`` function (accepting a target encoder model as the first argument) and the Python ``H2OTargetEncoderEstimator.transform`` methods are now fully compatible: they accept the same parameters and work consistently.
+- The parameters ``data_leakage_handling``, ``seed`` are now ignored on those methods: ``transform`` will use by default the corresponding values defined when building the TargetEncoder model.
```

Comment:
@sebhrusen what is the reasoning behind this, this one of Andrey's design goals to be able to change these parameters without re-training. I think dropping seed shouldn't hurt but what about the leakage handling?

Does this also affect SW API? If so how? cc: @mn-mikke (adding as a reviewer)

Reply:
From my understanding, the idea of providing `data_leakage_handling` in `transform` method was not to avoid re-training (we're still using the same encodings), but to allow a different encoding of training data vs. validation/test data.
Typically, when user wants to use a data leakage strategy, this is the for the training data:
- kfold requires a fold column, that you're unlikely to find in validation/test data, and that you surely don't have in production data.
- leave-one-out requires the target, that you may not have in test data, and that you surely don't have in production data.
So, I think the original approach was to expose those low-level parameters and let the user decide when to apply the `data_leakage_handling`, meaning that the user will have to deeply understand how it works to avoid mistakes.
For example if you trained your TE model with `kfold` strategy, then it doesn't really make sense to encode your training data with `leave-one-out` strategy, but it was still possible.
One the other way, if you trained your TE model with `leave-one-out` strategy, then you wouldn't be able to encode your data with `kfold`... same if you trained it with `none` strategy.

Due to these inconsistencies, I prefer to train the TE with one strategy, and then use the new param `as_training` in the `transform` methods, to use the strategy for training data, and disable it (as_training=false by default) for other data, focusing on user intent instead of technical parameters that require user to understand in detail what's happening.

Also note that by preventing use of `data_leakage_handling` in `transform`, it allows me to do some optimizations in the Java implementation: some groupings can be done during training instead of doing them each time we call `transform`...